### DISCUSSION TITLE
spdlog output to Serial

### DISCUSSION THREAD

**Discussion by kowalski100** [gabime/spdlog#3218] [General]
Hi,

I was exploring to use the spdlog library on some baremetal firmware without OS. Is it possible to use/redirect spdlog to serial port?
Our C++ version cannot go beyond C++11 due to low level requirements.

Thanks

**Comment by gabime** [comment]
Implementing a custom sink for this seems like a good option. Should be straight forward

**Reply by kowalski100** [reply]
Hi @gabime 

custom sink - you mean adding a glue logic b/w serial hardware and spdlog output?

**Comment by gabime** [comment]
Yes. See the docs on how to implement a custom sink (to send the log to a serial port).

**Reply by kowalski100** [reply]
sure. thanks for the help @gabime. 

I just have one more quick question for your expert opinion: Is there any advantage to using the spdlog library (with serial) beyond formatting text, since the same can be achieved with sprintf- like functions?

**Reply by gabime** [reply]
spdlog have many features. It all depends on the needs
